src/layers/repository/utils.py

```diff
@@ -0,0 +1,62 @@
+from enum import Enum
+
+from repository.errors import UnhandledException
+
+
+class Foo:
+    pass
+
+
+def to_dict(o):
+    """
+    Recursive __dict__ call
+    """
+    if isinstance(o, (int, float, bool, str, tuple, type, type(None))):
+        return o
+    elif isinstance(o, dict):
+        return {k: to_dict(v) for (k, v) in o.items()}
+    elif isinstance(o, list):
+        return [to_dict(v) for v in o]
+    return {k: to_dict(v) for (k, v) in o.__dict__.items() if not k.startswith("__")}
+
+
+def marshall_value(value):
+    if value is None:
+        return {"Null": True}
+    if isinstance(value, Enum):
```

Comment:
could fail here if the enum value is not a str - better use `StrEnum` all over the place